### ISSUE TITLE
check ticket.

### ISSUE THREAD

**Issue by csell5** [ThatConference/that.us#62]

**Comment by brettski** [comment]
A POC Checkin is created in branch `feature/ticket-check`

The function is located at /api/checkin.js and is executed using vercel goo. It uses Ti.to's checking functionality so no Ti.to api auth is required. There is one secret, checking list id, which needs to be handled which will be figured out depending on where this is run (vercel or gcp)

Function workflow:
- Accepts a GET request with a query parameter of `reference`. The ticket reference (e.g. ABCD-1) is passed in. The RegEx, `/^[a-zA-Z0-9]{4}-[0-9]/`, can be used for verifying the value from the user on the front end.
- A request is made against Ti.to to get a json collection of all tickets
- The passed in `ticket reference` is verified against the collection of tickets.
  - if **no match** return http 404 Not Found
  - if **Match > 1** return http 406 Not acceptable
- On a successful ticket match
  - Checking ticket at Ti.to adding the attendee to the checkin list
    - If checkin fails return http 502
  - Return http 200 and json response with ticket information. (need to determine what of this we wish to redact from output)

```json
{
  "id": 6022981,
  "slug": "ti_pfQQkuAW1CNnt2DboVkeisA",
  "first_name": "Brian",
  "last_name": "--lastname--",
  "email": "--email@address.ss",
  "phone_number": "11234567890",
  "company_name": "None",
  "release_title": "Patron Camper",
  "reference": "AA11-1",
  "registration_reference": "AA11",
  "tags": null,
  "created_at": "2020-07-14T19:21:19.000Z",
  "updated_at": "2020-07-14T19:24:06.000Z"
}
```

.
.
The checkin return the following json, besides having a date and id to validate checkin, the data isn't super useful
```json
{
  "id": 2019594,
  "checkin_list_id": 1016414,
  "ticket_id": 6022981,
  "created_at": "2020-07-17T20:07:47.000Z",
  "updated_at": "2020-07-17T20:07:47.000Z",
  "uuid_bin": null,
  "deleted_at": null,
  "uuid": "ed9c5602-2442-40d0-aa82-4a1802c454f9"
}
```

**Comment by csell5** [comment]
implemented. closed by #218
